### ISSUE TITLE
Optimize all parameters in Pytorch module that require gradient

### ISSUE THREAD

**Issue by JackCaster** [facebookresearch/nevergrad#1602]
I would like to optimize a Pytorch neural network with Nevergrad. How can I tell Nevergrad to optimize all parameters in the module that require grad (e.g., `params_to_optimize = filter(lambda p: p.requires_grad, model.parameters())`)?

**Comment by changdaeoh** [comment]
@JackCaster Hi

It's been a while, so you've probably resolved it on your own, but I found the tutorial below useful.

https://medium.com/@joragasy/optimize-neural-network-with-gradient-free-methods-using-pytorch-and-nevergrad-399a9f4a5c21

**Comment by JackCaster** [comment]
Any additional (official) inputs about this?

**Comment by teytaud** [comment]
I think we need something for conveniently optimizing a subset of a model parameters.
I do it everyday, but manually each time.
A convenient interface with Pytorch models would be great.
I will investigate. This will take some time though.
